README.md

```diff
@@ -1,27 +1,27 @@
+The app uses [Amazon Key Management Service (KMS)](https://aws.amazon.com/kms/) to create and control cryptographic keys and the [AWS SDK for Python (boto3)](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/kms.html) to call KMS. Note that this example can also run locally with [LocalStack](https://www.localstack.cloud/) used to simulate interactions with AWS.
 
 ### About CSRs
 
-In addition to being an example of using the Python library, this app shows how to generate a certificate signing request (CSR), a message sent to a certificate authority (CA) to request the signing of a public key and associated information. 
+In addition to being an example of using the Python library, this app shows how to generate a certificate signing request (CSR), a message sent to a certificate authority (CA) to request the signing of a public key and associated information.
 
-A CSR comprises a public key, as well as a common name, organization, city, state, country, and e-mail address. Not all of these fields are required and will vary depending with the assurance level of your certificate. 
+A CSR comprises a public key, as well as a common name, organization, city, state, country, and e-mail address. Not all of these fields are required and will vary depending with the assurance level of your certificate.
 
-You sign the CSR with your private key; this proves to the CA that you have control of the private key that corresponds to the public key included in the CSR. Once the requested information in a CSR passes a vetting process and domain control is established, the CA may sign the public key to indicate that it can be publicly trusted. 
+You sign the CSR with your private key; this proves to the CA that you have control of the private key that corresponds to the public key included in the CSR. Once the requested information in a CSR passes a vetting process and domain control is established, the CA may sign the public key to indicate that it can be publicly trusted.
 
 ## Prerequisites
 
 To build and run this app, you must install:
 
-- Python 3.10.
+- Python 3.10 or Python 3.12.
 - OpenSSL: See [OpenSSL](https://www.openssl.org/source/) for the source distribution or the [list of unofficial binary distributions](https://wiki.openssl.org/index.php/Binaries).  Make sure you have a recent version.
 
-You must also have an AWS account and be able to get standard AWS access credentials so you can use KMS.
+If you wish to run this example with AWS, you must also have an AWS account and be able to get standard AWS access credentials so you can use KMS. If you prefer to run everything locally instead, you can follow the steps for [LocalStack](https://www.localstack.cloud/) setup to run the example on your machine without accessing a real AWS environment.
```

Comment:
Suggest this change:

To run this example entirely locally for development and testing, follow the steps in [Using LocalStack](#using-localstack) to setup and run a mock AWS environment runing locally.